app/src/main/java/com/example/mcateam6/fragments/BarcodeFragment.kt

```diff
@@ -0,0 +1,117 @@
+package com.example.mcateam6.fragments
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import com.example.mcateam6.R
+import android.Manifest
+import android.content.pm.PackageManager
+import android.graphics.Matrix
+import android.util.Log
+import android.util.Rational
+import android.util.Size
+import android.view.*
+import android.widget.Toast
+import androidx.camera.core.CameraX
+import androidx.camera.core.Preview
+import androidx.camera.core.PreviewConfig
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import com.example.mcateam6.activities.MainActivity
+
+private const val REQUEST_CODE_PERMISSIONS = 10
+
+private val REQUIRED_PERMISSIONS = arrayOf(Manifest.permission.CAMERA)
+
+class BarcodeFragment : Fragment() {
+    private lateinit var view_finder: TextureView
```

Comment:
Not that important, but Android Studio complains about naming conventions (view_finder instead of viewFinder)